### PR TITLE
feature: updated Iroha2 to rc19 + refactoring in kotlin-query-tool

### PR DESCRIPTION
**Pull request by gv-timur**
updated Iroha2 to rc19 + refactoring

### CODE REVIEW

**Review by Mingela — APPROVED** [review]
great, thanks for the tests

**Review by Mingela — APPROVED** [review]

File: iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import java.security.KeyPair
+
+@Service
+class IrohaService {
+
+    fun query(config: Iroha2Config): String {
+        if (config.url.isEmpty() || config.adminAccount.isEmpty() ||
+            config.adminPublicKeyHex.isEmpty() || config.adminPrivateKeyHex.isEmpty()
+        ) {
+            throw RuntimeException("Incorrect config")
```
Comment (arndey) [inline iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt:29]: I'd separate this cases via `when` and throw exceptions with different messages

File: iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+import jp.co.soramitsu.iroha2.query.QueryBuilder
+import jp.co.soramitsu.iroha2.toHex
+import kotlinx.coroutines.runBlocking
+import org.springframework.stereotype.Service
+import java.security.KeyPair
+
+@Service
+class IrohaService {
+
+    fun query(config: Iroha2Config): String {
```
Comment (arndey) [inline iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt:25]: make is suspend and get rid of all `runBlocking`'s

File: iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                            )
+                        }
+                        val accountSignatories = account.signatories.map { publicKey ->
+                            Signature(publicKeyHex = publicKey.payload.toHex())
+                        }
+                        val accountMetadata = account.metadata.map.entries.associate {
+                            it.key.string to it.value.toString()
+                        }.toMap()
+
+                        val permissions = QueryBuilder.findPermissionTokensByAccountId(accountId)
```
Comment (arndey) [inline iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt:76]: could be extracted to a separate fun

File: iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+                            signatories = accountSignatories,
+                            metadata = accountMetadata,
+                            roles = account.roles.map { it.name.string },
+                            permissions
+                        )
+                    }
+                    val domainMetadata = domain.metadata.map.entries.associate {
+                        it.key.string to it.value.toString()
+                    }.toMap()
+                    val assetDefinitions = domain.assetDefinitions.map { (t, u) ->
```
Comment (arndey) [inline iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt:104]: could be extracted to a separate fun

File: iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+                    }
+                    Domain(
+                        id = domain.id.name.string,
+                        domainMetadata = domainMetadata,
+                        accounts = accounts,
+                        assetDefinitions = assetDefinitions
+                    )
+                }
+            }.let {
+                val mapper = ObjectMapper()
```
Comment (arndey) [inline iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt:132]: better to make it as class field

File: iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+        ) {
+            throw RuntimeException("Incorrect config")
+        }
+
+        val credentials = when (config.username.isNotEmpty() && config.password.isNotEmpty()) {
+            true -> "${config.username}:${config.password}"
+            false -> null
+        }
+
+        val iroha2Client = Iroha2Client(
```
Comment (arndey) [inline iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt:37]: why it's not a Spring bean?

**Review by arndey — CHANGES_REQUESTED** [review]
comments

File: iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt (hunk 5 above)
Comment (gv-timur) [reply iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt]: fixed

File: iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt (hunk 4 above)
Comment (gv-timur) [reply iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt]: done

File: iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt (hunk 3 above)
Comment (gv-timur) [reply iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt]: done

File: iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt (hunk 1 above)
Comment (gv-timur) [reply iroha2-kotlin-query-tool/src/main/kotlin/jp/co/genesis/service/IrohaService.kt]: done

**Review by arndey — APPROVED** [review]

**Pull request merged**